### PR TITLE
pyenv support

### PR DESCRIPTION
**Pull request by cr-mitmit**
Support for pyenv right prompt

Introduce _tide_item_pyenv.fish which extracts pyenv version python projects if pyenv is installed on machine

#### Motivation and Context

Closes # 

#### How Has This Been Tested

- [ ] I have tested using **Linux**.
- [ v] I have tested using **MacOS**.

#### Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: functions/_tide_item_pyenv.fish [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function _tide_item_pyenv
+    if test -n "$PYENV_VERSION" \
+            -o -f .python-version \
+            -o -f requirements.txt \
+            -o -f pyproject.toml \
+            -o (count *.py) -gt 0
+        set -l pyenv_status (pyenv version-name 2>/dev/null)
+        _tide_print_item pyenv $tide_pyenv_icon' ' $pyenv_status
+    end
+end
```
Comment (jorgebucaran) [inline functions/_tide_item_pyenv.fish:10]: Using Fish's `test` builtin like this has a catch. Even if `PYENV_VERSION` is set and not empty, every expression that follows will be evaluated up, which is not ideal.

Therefore, I think the following would be more efficient:

```fish
function _tide_item_pyenv
    test -n "$PYENV_VERSION" ||
        test -f .python-version ||
        test -f requirements.txt ||
        test -f pyproject.toml ||
        test (count *.py) -gt 0 &&
        _tide_print_item pyenv $tide_pyenv_icon' ' (pyenv version-name 2>/dev/null)
end
```

**Review by jorgebucaran — CHANGES_REQUESTED** [review]
@cr-mitmit I have a suggestion for improving the pyenv item. Everything else looks good to me. 👌

File: functions/_tide_item_pyenv.fish (hunk 1 above)
Comment (IlanCosman) [reply functions/_tide_item_pyenv.fish:10]: I think actually the new `path` builtin would be best for this. But I'd like to combine this with the `venv` item into a new `python` item for Tide v6.

Comment (jorgebucaran) [reply functions/_tide_item_pyenv.fish:10]: Even better. 👍

**Comment by IlanCosman** [comment]
I think the important functionality of this should be done in the new `python` item, if not, let me know :)

**Pull request closed without merging**